src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -4361,6 +4487,13 @@ protected virtual void Dispose(bool disposing)
 
             SetToDisconnected();
 
+            // If the safe handle doesn't own the underlying handle, we're done.
+            SafeSocketHandle handle = _handle;
+            if (handle != null && !handle.OwnsHandle)
+            {
+                return;
+            }
```

Comment:
@tmds, I made Socket.Dispose a (mostly) nop when the SafeHandle doesn't own the underlying file descriptor / handle.